File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Visitor/Section.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Visitor {
    internal class Section : IVisited {
        private readonly Title title;
        private readonly List<Paragraph> paragraphs = new List<Paragraph>();

        public void AcceptVisitor(IVisitor v) {
            v.VisitSectionStart(this);
            title.AcceptVisitor(v);
            foreach (var p in paragraphs) {
                p.AcceptVisitor(v);
            }
            v.VisitSectionEnd();
        }

        public Section(Title title) {
            this.title = title;
        }

        public void AddParagraph(Paragraph p) {
            paragraphs.Add(p);
        }
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/visitor/Section.java
package patterns.visitor;

import java.util.*;

public class Section implements Visited {
	@SuppressWarnings("unused")
	private Title title;
	private List<Paragraph> paragraphs = new ArrayList<Paragraph>();
	
	public void acceptVisitor(Visitor v) {
		v.visitSectionStart(this);
		title.acceptVisitor(v);
		for(Paragraph p : paragraphs) {
			p.acceptVisitor(v);
		}
		v.visitSectionEnd();
	}
	public Section(Title title) {
		this.title = title;
	}
	public void addParagraph(Paragraph p) {
		paragraphs.add(p);
	}
}
